// ng2-bootstrap/components/dropdown/dropdown-toggle.directive.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
import * as import0 from 'ng2-bootstrap/components/dropdown/dropdown-toggle.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
export var Wrapper_DropdownToggleDirective = (function () {
    function Wrapper_DropdownToggleDirective(p0, p1) {
        this.changed = false;
        this._isDisabled = import1.UNINITIALIZED;
        this._addToggleClass = import1.UNINITIALIZED;
        this.context = new import0.DropdownToggleDirective(p0, p1);
    }
    Wrapper_DropdownToggleDirective.prototype.check_isDisabled = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._isDisabled, currValue))) {
            this.changed = true;
            this.context.isDisabled = currValue;
            this._isDisabled = currValue;
        }
    };
    Wrapper_DropdownToggleDirective.prototype.check_addToggleClass = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._addToggleClass, currValue))) {
            this.changed = true;
            this.context.addToggleClass = currValue;
            this._addToggleClass = currValue;
        }
    };
    Wrapper_DropdownToggleDirective.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        if (!throwOnChange) {
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    return Wrapper_DropdownToggleDirective;
}());
