// ng2-bootstrap/components/progressbar/bar.component.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/progressbar/bar.component';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
import * as import4 from '@angular/core/src/render/api';
import * as import5 from '@angular/core/src/linker/element';
import * as import6 from '@angular/core/src/di/injector';
import * as import7 from '@angular/core/src/linker/view_type';
import * as import8 from 'ng2-bootstrap/components/progressbar/progress.directive';
import * as import9 from '@angular/core/src/metadata/view';
import * as import10 from '@angular/core/src/linker/component_factory';
import * as import11 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import12 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import13 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import14 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import15 from '@angular/core/src/linker/element_ref';
import * as import16 from '@angular/common/src/directives/ng_class';
import * as import17 from '@angular/common/src/directives/ng_style';
export class Wrapper_BarComponent {
  context:import0.BarComponent;
  changed:boolean;
  /*private*/ _type:any;
  /*private*/ _value:any;
  constructor(p0:any) {
    this.changed = false;
    this._type = import1.UNINITIALIZED;
    this._value = import1.UNINITIALIZED;
    this.context = new import0.BarComponent(p0);
  }
  check_type(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._type,currValue))) {
      this.changed = true;
      this.context.type = currValue;
      this._type = currValue;
    }
  }
  check_value(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._value,currValue))) {
      this.changed = true;
      this.context.value = currValue;
      this._value = currValue;
    }
  }
  detectChangesInternal(view:import3.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
}
var renderType_BarComponent_Host:import4.RenderComponentType = (null as any);
class _View_BarComponent_Host0 extends import3.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import5.AppElement;
  _BarComponent_0_4:Wrapper_BarComponent;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import6.Injector,declarationEl:import5.AppElement) {
    super(_View_BarComponent_Host0,renderType_BarComponent_Host,import7.ViewType.HOST,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.selectOrCreateHostElement('bar',rootSelector,(null as any));
    this._appEl_0 = new import5.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_BarComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._BarComponent_0_4 = new Wrapper_BarComponent(this.parentInjector.get(import8.ProgressDirective));
    this._appEl_0.initComponent(this._BarComponent_0_4.context,([] as any[]),compView_0);
    compView_0.create(this._BarComponent_0_4.context,this.projectableNodes,(null as any));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.BarComponent) && (0 === requestNodeIndex))) { return this._BarComponent_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._BarComponent_0_4.detectChangesInternal(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._BarComponent_0_4.context.ngOnDestroy();
  }
}
function viewFactory_BarComponent_Host0(viewUtils:import2.ViewUtils,parentInjector:import6.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  if ((renderType_BarComponent_Host === (null as any))) { (renderType_BarComponent_Host = viewUtils.createRenderComponentType('',0,import9.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_BarComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const BarComponentNgFactory:import10.ComponentFactory<import0.BarComponent> = new import10.ComponentFactory<import0.BarComponent>('bar',viewFactory_BarComponent_Host0,import0.BarComponent);
const styles_BarComponent:any[] = ([] as any[]);
var renderType_BarComponent:import4.RenderComponentType = (null as any);
class _View_BarComponent0 extends import3.AppView<import0.BarComponent> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import11.Wrapper_NgClass;
  _NgStyle_1_4:import12.Wrapper_NgStyle;
  _text_2:any;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  _map_0:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import6.Injector,declarationEl:import5.AppElement) {
    super(_View_BarComponent0,renderType_BarComponent,import7.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this._el_1 = this.renderer.createElement(parentRenderNode,'div',(null as any));
    this.renderer.setElementAttribute(this._el_1,'aria-valuemin','0');
    this.renderer.setElementAttribute(this._el_1,'class','progress-bar');
    this.renderer.setElementAttribute(this._el_1,'role','progressbar');
    this.renderer.setElementAttribute(this._el_1,'style','min-width: 0;');
    this._NgClass_1_3 = new import11.Wrapper_NgClass(this.parentInjector.get(import13.IterableDiffers),this.parentInjector.get(import14.KeyValueDiffers),new import15.ElementRef(this._el_1),this.renderer);
    this._NgStyle_1_4 = new import12.Wrapper_NgStyle(this.parentInjector.get(import14.KeyValueDiffers),new import15.ElementRef(this._el_1),this.renderer);
    this.renderer.projectNodes(this._el_1,import2.flattenNestedViewRenderNodes(this.projectableNodes[0]));
    this._text_2 = this.renderer.createText(parentRenderNode,'\n',(null as any));
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    this._map_0 = import2.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        width: p0,
        transition: p1
      }
      ;
    });
    this.init(([] as any[]),[
      this._text_0,
      this._el_1,
      this._text_2
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.NgClass) && (1 === requestNodeIndex))) { return this._NgClass_1_3.context; }
    if (((token === import17.NgStyle) && (1 === requestNodeIndex))) { return this._NgStyle_1_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_3:any = 'progress-bar';
    this._NgClass_1_3.check_klass(currVal_3,throwOnChange,false);
    const currVal_4:any = (this.context.type && ('progress-bar-' + this.context.type));
    this._NgClass_1_3.check_ngClass(currVal_4,throwOnChange,false);
    this._NgClass_1_3.detectChangesInternal(this,this._el_1,throwOnChange);
    const currVal_5:any = this._map_0((((this.context.percent < 100)? this.context.percent: 100) + '%'),this.context.transition);
    this._NgStyle_1_4.check_ngStyle(currVal_5,throwOnChange,false);
    this._NgStyle_1_4.detectChangesInternal(this,this._el_1,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = this.context.value;
    if (import2.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setElementAttribute(this._el_1,'aria-valuenow',((currVal_0 == (null as any))? (null as any): currVal_0.toString()));
      this._expr_0 = currVal_0;
    }
    const currVal_1:any = (this.context.percent.toFixed(0) + '%');
    if (import2.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementAttribute(this._el_1,'aria-valuetext',((currVal_1 == (null as any))? (null as any): currVal_1.toString()));
      this._expr_1 = currVal_1;
    }
    const currVal_2:any = this.context.max;
    if (import2.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementAttribute(this._el_1,'aria-valuemax',((currVal_2 == (null as any))? (null as any): currVal_2.toString()));
      this._expr_2 = currVal_2;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
}
export function viewFactory_BarComponent0(viewUtils:import2.ViewUtils,parentInjector:import6.Injector,declarationEl:import5.AppElement):import3.AppView<import0.BarComponent> {
  if ((renderType_BarComponent === (null as any))) { (renderType_BarComponent = viewUtils.createRenderComponentType('',1,import9.ViewEncapsulation.None,styles_BarComponent,{})); }
  return new _View_BarComponent0(viewUtils,parentInjector,declarationEl);
}
