// ng2-bootstrap/components/progressbar/progress.directive.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/progressbar/progress.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
export class Wrapper_ProgressDirective {
  context:import0.ProgressDirective;
  changed:boolean;
  /*private*/ _animate:any;
  /*private*/ _max:any;
  constructor() {
    this.changed = false;
    this._animate = import1.UNINITIALIZED;
    this._max = import1.UNINITIALIZED;
    this.context = new import0.ProgressDirective();
  }
  check_animate(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._animate,currValue))) {
      this.changed = true;
      this.context.animate = currValue;
      this._animate = currValue;
    }
  }
  check_max(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._max,currValue))) {
      this.changed = true;
      this.context.max = currValue;
      this._max = currValue;
    }
  }
  detectChangesInternal(view:import3.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
}
