// ng2-bootstrap/components/buttons/button-radio.directive.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/buttons/button-radio.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
export class Wrapper_ButtonRadioDirective {
  context:import0.ButtonRadioDirective;
  changed:boolean;
  /*private*/ _btnRadio:any;
  /*private*/ _uncheckable:any;
  /*private*/ _value:any;
  constructor(p0:any) {
    this.changed = false;
    this._btnRadio = import1.UNINITIALIZED;
    this._uncheckable = import1.UNINITIALIZED;
    this._value = import1.UNINITIALIZED;
    this.context = new import0.ButtonRadioDirective(p0);
  }
  check_btnRadio(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._btnRadio,currValue))) {
      this.changed = true;
      this.context.btnRadio = currValue;
      this._btnRadio = currValue;
    }
  }
  check_uncheckable(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._uncheckable,currValue))) {
      this.changed = true;
      this.context.uncheckable = currValue;
      this._uncheckable = currValue;
    }
  }
  check_value(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._value,currValue))) {
      this.changed = true;
      this.context.value = currValue;
      this._value = currValue;
    }
  }
  detectChangesInternal(view:import3.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
}
