// ng2-bootstrap/components/buttons/button-checkbox.directive.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/buttons/button-checkbox.directive';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
export class Wrapper_ButtonCheckboxDirective {
  context:import0.ButtonCheckboxDirective;
  changed:boolean;
  /*private*/ _btnCheckboxTrue:any;
  /*private*/ _btnCheckboxFalse:any;
  constructor(p0:any) {
    this.changed = false;
    this._btnCheckboxTrue = import1.UNINITIALIZED;
    this._btnCheckboxFalse = import1.UNINITIALIZED;
    this.context = new import0.ButtonCheckboxDirective(p0);
  }
  check_btnCheckboxTrue(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._btnCheckboxTrue,currValue))) {
      this.changed = true;
      this.context.btnCheckboxTrue = currValue;
      this._btnCheckboxTrue = currValue;
    }
  }
  check_btnCheckboxFalse(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._btnCheckboxFalse,currValue))) {
      this.changed = true;
      this.context.btnCheckboxFalse = currValue;
      this._btnCheckboxFalse = currValue;
    }
  }
  detectChangesInternal(view:import3.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
}
