// ng2-bootstrap/components/pagination/pagination.component.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/pagination/pagination.component';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
import * as import4 from '@angular/core/src/render/api';
import * as import5 from '@angular/core/src/linker/element';
import * as import6 from '@angular/forms/src/directives/ng_model';
import * as import7 from '@angular/core/src/di/injector';
import * as import8 from '@angular/core/src/linker/view_type';
import * as import9 from '@angular/forms/src/directives/control_container';
import * as import10 from '@angular/core/src/linker/element_ref';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from '@angular/core/src/linker/component_factory';
import * as import13 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import14 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import15 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import16 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import17 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import18 from '@angular/core/src/linker/template_ref';
import * as import19 from '@angular/common/src/directives/ng_if';
import * as import20 from '@angular/common/src/directives/ng_for';
import * as import21 from '@angular/common/src/directives/ng_class';
import * as import22 from '@angular/core/src/security';
export class Wrapper_PaginationComponent {
  context:import0.PaginationComponent;
  changed:boolean;
  /*private*/ _align:any;
  /*private*/ _maxSize:any;
  /*private*/ _boundaryLinks:any;
  /*private*/ _directionLinks:any;
  /*private*/ _firstText:any;
  /*private*/ _previousText:any;
  /*private*/ _nextText:any;
  /*private*/ _lastText:any;
  /*private*/ _rotate:any;
  /*private*/ _pageBtnClass:any;
  /*private*/ _disabled:any;
  /*private*/ _itemsPerPage:any;
  /*private*/ _totalItems:any;
  constructor(p0:any,p1:any,p2:any) {
    this.changed = false;
    this._align = import1.UNINITIALIZED;
    this._maxSize = import1.UNINITIALIZED;
    this._boundaryLinks = import1.UNINITIALIZED;
    this._directionLinks = import1.UNINITIALIZED;
    this._firstText = import1.UNINITIALIZED;
    this._previousText = import1.UNINITIALIZED;
    this._nextText = import1.UNINITIALIZED;
    this._lastText = import1.UNINITIALIZED;
    this._rotate = import1.UNINITIALIZED;
    this._pageBtnClass = import1.UNINITIALIZED;
    this._disabled = import1.UNINITIALIZED;
    this._itemsPerPage = import1.UNINITIALIZED;
    this._totalItems = import1.UNINITIALIZED;
    this.context = new import0.PaginationComponent(p0,p1,p2);
  }
  check_align(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._align,currValue))) {
      this.changed = true;
      this.context.align = currValue;
      this._align = currValue;
    }
  }
  check_maxSize(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._maxSize,currValue))) {
      this.changed = true;
      this.context.maxSize = currValue;
      this._maxSize = currValue;
    }
  }
  check_boundaryLinks(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._boundaryLinks,currValue))) {
      this.changed = true;
      this.context.boundaryLinks = currValue;
      this._boundaryLinks = currValue;
    }
  }
  check_directionLinks(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._directionLinks,currValue))) {
      this.changed = true;
      this.context.directionLinks = currValue;
      this._directionLinks = currValue;
    }
  }
  check_firstText(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._firstText,currValue))) {
      this.changed = true;
      this.context.firstText = currValue;
      this._firstText = currValue;
    }
  }
  check_previousText(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._previousText,currValue))) {
      this.changed = true;
      this.context.previousText = currValue;
      this._previousText = currValue;
    }
  }
  check_nextText(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._nextText,currValue))) {
      this.changed = true;
      this.context.nextText = currValue;
      this._nextText = currValue;
    }
  }
  check_lastText(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._lastText,currValue))) {
      this.changed = true;
      this.context.lastText = currValue;
      this._lastText = currValue;
    }
  }
  check_rotate(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._rotate,currValue))) {
      this.changed = true;
      this.context.rotate = currValue;
      this._rotate = currValue;
    }
  }
  check_pageBtnClass(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._pageBtnClass,currValue))) {
      this.changed = true;
      this.context.pageBtnClass = currValue;
      this._pageBtnClass = currValue;
    }
  }
  check_disabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._disabled,currValue))) {
      this.changed = true;
      this.context.disabled = currValue;
      this._disabled = currValue;
    }
  }
  check_itemsPerPage(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._itemsPerPage,currValue))) {
      this.changed = true;
      this.context.itemsPerPage = currValue;
      this._itemsPerPage = currValue;
    }
  }
  check_totalItems(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._totalItems,currValue))) {
      this.changed = true;
      this.context.totalItems = currValue;
      this._totalItems = currValue;
    }
  }
  detectChangesInternal(view:import3.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
}
var renderType_PaginationComponent_Host:import4.RenderComponentType = (null as any);
class _View_PaginationComponent_Host0 extends import3.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import5.AppElement;
  _NgModel_0_4:import6.NgModel;
  _PaginationComponent_0_5:Wrapper_PaginationComponent;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_PaginationComponent_Host0,renderType_PaginationComponent_Host,import8.ViewType.HOST,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.selectOrCreateHostElement('pagination',rootSelector,(null as any));
    this.renderer.setElementAttribute(this._el_0,'ngModel','');
    this._appEl_0 = new import5.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_PaginationComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._NgModel_0_4 = new import6.NgModel(this.parentInjector.get(import9.ControlContainer,(null as any)),(null as any),(null as any),(null as any));
    this._PaginationComponent_0_5 = new Wrapper_PaginationComponent(this._NgModel_0_4,this.renderer,new import10.ElementRef(this._el_0));
    this._appEl_0.initComponent(this._PaginationComponent_0_5.context,([] as any[]),compView_0);
    compView_0.create(this._PaginationComponent_0_5.context,this.projectableNodes,(null as any));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import6.NgModel) && (0 === requestNodeIndex))) { return this._NgModel_0_4; }
    if (((token === import0.PaginationComponent) && (0 === requestNodeIndex))) { return this._PaginationComponent_0_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._PaginationComponent_0_5.detectChangesInternal(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._NgModel_0_4.ngOnDestroy();
  }
}
function viewFactory_PaginationComponent_Host0(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  if ((renderType_PaginationComponent_Host === (null as any))) { (renderType_PaginationComponent_Host = viewUtils.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_PaginationComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const PaginationComponentNgFactory:import12.ComponentFactory<import0.PaginationComponent> = new import12.ComponentFactory<import0.PaginationComponent>('pagination[ngModel]',viewFactory_PaginationComponent_Host0,import0.PaginationComponent);
const styles_PaginationComponent:any[] = ([] as any[]);
var renderType_PaginationComponent:import4.RenderComponentType = (null as any);
class _View_PaginationComponent0 extends import3.AppView<import0.PaginationComponent> {
  _text_0:any;
  _el_1:any;
  _NgClass_1_3:import13.Wrapper_NgClass;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _appEl_3:import5.AppElement;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import14.Wrapper_NgIf;
  _text_4:any;
  _anchor_5:any;
  /*private*/ _appEl_5:import5.AppElement;
  _TemplateRef_5_5:any;
  _NgIf_5_6:import14.Wrapper_NgIf;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _appEl_7:import5.AppElement;
  _TemplateRef_7_5:any;
  _NgFor_7_6:import15.Wrapper_NgFor;
  _text_8:any;
  _anchor_9:any;
  /*private*/ _appEl_9:import5.AppElement;
  _TemplateRef_9_5:any;
  _NgIf_9_6:import14.Wrapper_NgIf;
  _text_10:any;
  _anchor_11:any;
  /*private*/ _appEl_11:import5.AppElement;
  _TemplateRef_11_5:any;
  _NgIf_11_6:import14.Wrapper_NgIf;
  _text_12:any;
  _text_13:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_PaginationComponent0,renderType_PaginationComponent,import8.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this._el_1 = this.renderer.createElement(parentRenderNode,'ul',(null as any));
    this.renderer.setElementAttribute(this._el_1,'class','pagination');
    this._NgClass_1_3 = new import13.Wrapper_NgClass(this.parentInjector.get(import16.IterableDiffers),this.parentInjector.get(import17.KeyValueDiffers),new import10.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n    ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._appEl_3 = new import5.AppElement(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import18.TemplateRef_(this._appEl_3,viewFactory_PaginationComponent1);
    this._NgIf_3_6 = new import14.Wrapper_NgIf(this._appEl_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText(this._el_1,'\n\n    ',(null as any));
    this._anchor_5 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._appEl_5 = new import5.AppElement(5,1,this,this._anchor_5);
    this._TemplateRef_5_5 = new import18.TemplateRef_(this._appEl_5,viewFactory_PaginationComponent2);
    this._NgIf_5_6 = new import14.Wrapper_NgIf(this._appEl_5.vcRef,this._TemplateRef_5_5);
    this._text_6 = this.renderer.createText(this._el_1,'\n\n    ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._appEl_7 = new import5.AppElement(7,1,this,this._anchor_7);
    this._TemplateRef_7_5 = new import18.TemplateRef_(this._appEl_7,viewFactory_PaginationComponent3);
    this._NgFor_7_6 = new import15.Wrapper_NgFor(this._appEl_7.vcRef,this._TemplateRef_7_5,this.parentInjector.get(import16.IterableDiffers),this.ref);
    this._text_8 = this.renderer.createText(this._el_1,'\n\n    ',(null as any));
    this._anchor_9 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._appEl_9 = new import5.AppElement(9,1,this,this._anchor_9);
    this._TemplateRef_9_5 = new import18.TemplateRef_(this._appEl_9,viewFactory_PaginationComponent4);
    this._NgIf_9_6 = new import14.Wrapper_NgIf(this._appEl_9.vcRef,this._TemplateRef_9_5);
    this._text_10 = this.renderer.createText(this._el_1,'\n\n    ',(null as any));
    this._anchor_11 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._appEl_11 = new import5.AppElement(11,1,this,this._anchor_11);
    this._TemplateRef_11_5 = new import18.TemplateRef_(this._appEl_11,viewFactory_PaginationComponent5);
    this._NgIf_11_6 = new import14.Wrapper_NgIf(this._appEl_11.vcRef,this._TemplateRef_11_5);
    this._text_12 = this.renderer.createText(this._el_1,'\n  ',(null as any));
    this._text_13 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this.init(([] as any[]),[
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._anchor_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._anchor_9,
      this._text_10,
      this._anchor_11,
      this._text_12,
      this._text_13
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import18.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import19.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    if (((token === import18.TemplateRef) && (5 === requestNodeIndex))) { return this._TemplateRef_5_5; }
    if (((token === import19.NgIf) && (5 === requestNodeIndex))) { return this._NgIf_5_6.context; }
    if (((token === import18.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import20.NgFor) && (7 === requestNodeIndex))) { return this._NgFor_7_6.context; }
    if (((token === import18.TemplateRef) && (9 === requestNodeIndex))) { return this._TemplateRef_9_5; }
    if (((token === import19.NgIf) && (9 === requestNodeIndex))) { return this._NgIf_9_6.context; }
    if (((token === import18.TemplateRef) && (11 === requestNodeIndex))) { return this._TemplateRef_11_5; }
    if (((token === import19.NgIf) && (11 === requestNodeIndex))) { return this._NgIf_11_6.context; }
    if (((token === import21.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 12)))) { return this._NgClass_1_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0:any = 'pagination';
    this._NgClass_1_3.check_klass(currVal_0,throwOnChange,false);
    const currVal_1:any = this.context.classMap;
    this._NgClass_1_3.check_ngClass(currVal_1,throwOnChange,false);
    this._NgClass_1_3.detectChangesInternal(this,this._el_1,throwOnChange);
    const currVal_2:any = this.context.boundaryLinks;
    this._NgIf_3_6.check_ngIf(currVal_2,throwOnChange,false);
    this._NgIf_3_6.detectChangesInternal(this,this._anchor_3,throwOnChange);
    const currVal_3:any = this.context.directionLinks;
    this._NgIf_5_6.check_ngIf(currVal_3,throwOnChange,false);
    this._NgIf_5_6.detectChangesInternal(this,this._anchor_5,throwOnChange);
    const currVal_4:any = this.context.pages;
    this._NgFor_7_6.check_ngForOf(currVal_4,throwOnChange,false);
    this._NgFor_7_6.detectChangesInternal(this,this._anchor_7,throwOnChange);
    const currVal_5:any = this.context.directionLinks;
    this._NgIf_9_6.check_ngIf(currVal_5,throwOnChange,false);
    this._NgIf_9_6.detectChangesInternal(this,this._anchor_9,throwOnChange);
    const currVal_6:any = this.context.boundaryLinks;
    this._NgIf_11_6.check_ngIf(currVal_6,throwOnChange,false);
    this._NgIf_11_6.detectChangesInternal(this,this._anchor_11,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
export function viewFactory_PaginationComponent0(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<import0.PaginationComponent> {
  if ((renderType_PaginationComponent === (null as any))) { (renderType_PaginationComponent = viewUtils.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_PaginationComponent,{})); }
  return new _View_PaginationComponent0(viewUtils,parentInjector,declarationEl);
}
class _View_PaginationComponent1 extends import3.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  /*private*/ _expr_0:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_PaginationComponent1,renderType_PaginationComponent,import8.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'li',(null as any));
    this.renderer.setElementAttribute(this._el_0,'class','pagination-first page-item');
    this._text_1 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this._el_2 = this.renderer.createElement(this._el_0,'a',(null as any));
    this.renderer.setElementAttribute(this._el_2,'class','page-link');
    this.renderer.setElementAttribute(this._el_2,'href','');
    this._text_3 = this.renderer.createText(this._el_0,'\n    ',(null as any));
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    var disposable_0:Function = this.renderer.listen(this._el_2,'click',this.eventHandler(this._handle_click_2_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = (this.parent.context.noPrevious() || this.parent.context.disabled);
    if (import2.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setElementClass(this._el_0,'disabled',currVal_0);
      this._expr_0 = currVal_0;
    }
    const currVal_2:any = this.parent.context.getText('first');
    if (import2.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_2,'innerHTML',this.viewUtils.sanitizer.sanitize(import22.SecurityContext.HTML,currVal_2));
      this._expr_2 = currVal_2;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_2_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.parent.context.selectPage(1,$event)) !== false);
    return (true && pd_0);
  }
}
function viewFactory_PaginationComponent1(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  return new _View_PaginationComponent1(viewUtils,parentInjector,declarationEl);
}
class _View_PaginationComponent2 extends import3.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  /*private*/ _expr_0:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_PaginationComponent2,renderType_PaginationComponent,import8.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'li',(null as any));
    this.renderer.setElementAttribute(this._el_0,'class','pagination-prev page-item');
    this._text_1 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this._el_2 = this.renderer.createElement(this._el_0,'a',(null as any));
    this.renderer.setElementAttribute(this._el_2,'class','page-link');
    this.renderer.setElementAttribute(this._el_2,'href','');
    this._text_3 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    var disposable_0:Function = this.renderer.listen(this._el_2,'click',this.eventHandler(this._handle_click_2_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = (this.parent.context.noPrevious() || this.parent.context.disabled);
    if (import2.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setElementClass(this._el_0,'disabled',currVal_0);
      this._expr_0 = currVal_0;
    }
    const currVal_2:any = this.parent.context.getText('previous');
    if (import2.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_2,'innerHTML',this.viewUtils.sanitizer.sanitize(import22.SecurityContext.HTML,currVal_2));
      this._expr_2 = currVal_2;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_2_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.parent.context.selectPage((this.parent.context.page - 1),$event)) !== false);
    return (true && pd_0);
  }
}
function viewFactory_PaginationComponent2(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  return new _View_PaginationComponent2(viewUtils,parentInjector,declarationEl);
}
class _View_PaginationComponent3 extends import3.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_3:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_PaginationComponent3,renderType_PaginationComponent,import8.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'li',(null as any));
    this.renderer.setElementAttribute(this._el_0,'class','pagination-page page-item');
    this._text_1 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this._el_2 = this.renderer.createElement(this._el_0,'a',(null as any));
    this.renderer.setElementAttribute(this._el_2,'class','page-link');
    this.renderer.setElementAttribute(this._el_2,'href','');
    this._text_3 = this.renderer.createText(this._el_0,'\n    ',(null as any));
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
    this._expr_3 = import1.UNINITIALIZED;
    var disposable_0:Function = this.renderer.listen(this._el_2,'click',this.eventHandler(this._handle_click_2_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = this.context.$implicit.active;
    if (import2.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setElementClass(this._el_0,'active',currVal_0);
      this._expr_0 = currVal_0;
    }
    const currVal_1:any = (this.parent.context.disabled && !this.context.$implicit.active);
    if (import2.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementClass(this._el_0,'disabled',currVal_1);
      this._expr_1 = currVal_1;
    }
    const currVal_3:any = this.context.$implicit.text;
    if (import2.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_2,'innerHTML',this.viewUtils.sanitizer.sanitize(import22.SecurityContext.HTML,currVal_3));
      this._expr_3 = currVal_3;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_2_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.parent.context.selectPage(this.context.$implicit.number,$event)) !== false);
    return (true && pd_0);
  }
}
function viewFactory_PaginationComponent3(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  return new _View_PaginationComponent3(viewUtils,parentInjector,declarationEl);
}
class _View_PaginationComponent4 extends import3.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  /*private*/ _expr_0:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_PaginationComponent4,renderType_PaginationComponent,import8.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'li',(null as any));
    this.renderer.setElementAttribute(this._el_0,'class','pagination-next page-item');
    this._text_1 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this._el_2 = this.renderer.createElement(this._el_0,'a',(null as any));
    this.renderer.setElementAttribute(this._el_2,'class','page-link');
    this.renderer.setElementAttribute(this._el_2,'href','');
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    var disposable_0:Function = this.renderer.listen(this._el_2,'click',this.eventHandler(this._handle_click_2_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = (this.parent.context.noNext() || this.parent.context.disabled);
    if (import2.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setElementClass(this._el_0,'disabled',currVal_0);
      this._expr_0 = currVal_0;
    }
    const currVal_2:any = this.parent.context.getText('next');
    if (import2.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_2,'innerHTML',this.viewUtils.sanitizer.sanitize(import22.SecurityContext.HTML,currVal_2));
      this._expr_2 = currVal_2;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_2_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.parent.context.selectPage((this.parent.context.page + 1),$event)) !== false);
    return (true && pd_0);
  }
}
function viewFactory_PaginationComponent4(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  return new _View_PaginationComponent4(viewUtils,parentInjector,declarationEl);
}
class _View_PaginationComponent5 extends import3.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  /*private*/ _expr_0:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_PaginationComponent5,renderType_PaginationComponent,import8.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'li',(null as any));
    this.renderer.setElementAttribute(this._el_0,'class','pagination-last page-item');
    this._text_1 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this._el_2 = this.renderer.createElement(this._el_0,'a',(null as any));
    this.renderer.setElementAttribute(this._el_2,'class','page-link');
    this.renderer.setElementAttribute(this._el_2,'href','');
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    var disposable_0:Function = this.renderer.listen(this._el_2,'click',this.eventHandler(this._handle_click_2_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = (this.parent.context.noNext() || this.parent.context.disabled);
    if (import2.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setElementClass(this._el_0,'disabled',currVal_0);
      this._expr_0 = currVal_0;
    }
    const currVal_2:any = this.parent.context.getText('last');
    if (import2.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementProperty(this._el_2,'innerHTML',this.viewUtils.sanitizer.sanitize(import22.SecurityContext.HTML,currVal_2));
      this._expr_2 = currVal_2;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_2_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.parent.context.selectPage(this.parent.context.totalPages,$event)) !== false);
    return (true && pd_0);
  }
}
function viewFactory_PaginationComponent5(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  return new _View_PaginationComponent5(viewUtils,parentInjector,declarationEl);
}
