// ng2-bootstrap/components/typeahead/typeahead-container.component.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from 'ng2-bootstrap/components/typeahead/typeahead-container.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/element';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/element_ref';
import * as import9 from 'ng2-bootstrap/components/typeahead/typeahead-options.class';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/component_factory';
import * as import12 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import13 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import14 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import15 from '@angular/core/src/linker/template_ref';
import * as import16 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import17 from '@angular/common/src/directives/ng_for';
import * as import18 from '@angular/common/src/directives/ng_style';
import * as import19 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import20 from '@angular/common/src/directives/ng_if';
import * as import21 from '@angular/core/src/security';
import * as import22 from '../../../@angular/common/src/directives/ng_template_outlet.ngfactory';
import * as import23 from '@angular/common/src/directives/ng_template_outlet';
export var Wrapper_TypeaheadContainerComponent = (function () {
    function Wrapper_TypeaheadContainerComponent(p0, p1) {
        this.changed = false;
        this.context = new import0.TypeaheadContainerComponent(p0, p1);
    }
    Wrapper_TypeaheadContainerComponent.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_TypeaheadContainerComponent;
}());
var renderType_TypeaheadContainerComponent_Host = null;
var _View_TypeaheadContainerComponent_Host0 = (function (_super) {
    __extends(_View_TypeaheadContainerComponent_Host0, _super);
    function _View_TypeaheadContainerComponent_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_TypeaheadContainerComponent_Host0, renderType_TypeaheadContainerComponent_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_TypeaheadContainerComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('typeahead-container', rootSelector, null);
        this._appEl_0 = new import3.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_TypeaheadContainerComponent0(this.viewUtils, this.injector(0), this._appEl_0);
        this._TypeaheadContainerComponent_0_4 = new Wrapper_TypeaheadContainerComponent(new import8.ElementRef(this._el_0), this.parentInjector.get(import9.TypeaheadOptions));
        this._appEl_0.initComponent(this._TypeaheadContainerComponent_0_4.context, [], compView_0);
        compView_0.create(this._TypeaheadContainerComponent_0_4.context, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_TypeaheadContainerComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.TypeaheadContainerComponent) && (0 === requestNodeIndex))) {
            return this._TypeaheadContainerComponent_0_4.context;
        }
        return notFoundResult;
    };
    _View_TypeaheadContainerComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._TypeaheadContainerComponent_0_4.detectChangesInternal(this, this._el_0, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_TypeaheadContainerComponent_Host0;
}(import1.AppView));
function viewFactory_TypeaheadContainerComponent_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_TypeaheadContainerComponent_Host === null)) {
        (renderType_TypeaheadContainerComponent_Host = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, [], {}));
    }
    return new _View_TypeaheadContainerComponent_Host0(viewUtils, parentInjector, declarationEl);
}
export var TypeaheadContainerComponentNgFactory = new import11.ComponentFactory('typeahead-container', viewFactory_TypeaheadContainerComponent_Host0, import0.TypeaheadContainerComponent);
var styles_TypeaheadContainerComponent = [];
var renderType_TypeaheadContainerComponent = null;
var _View_TypeaheadContainerComponent0 = (function (_super) {
    __extends(_View_TypeaheadContainerComponent0, _super);
    function _View_TypeaheadContainerComponent0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_TypeaheadContainerComponent0, renderType_TypeaheadContainerComponent, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_TypeaheadContainerComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n  ', null);
        this._el_1 = this.renderer.createElement(parentRenderNode, 'ul', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'dropdown-menu');
        this._NgStyle_1_3 = new import12.Wrapper_NgStyle(this.parentInjector.get(import14.KeyValueDiffers), new import8.ElementRef(this._el_1), this.renderer);
        this._text_2 = this.renderer.createText(this._el_1, '\n    ', null);
        this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1, null);
        this._appEl_3 = new import3.AppElement(3, 1, this, this._anchor_3);
        this._TemplateRef_3_5 = new import15.TemplateRef_(this._appEl_3, viewFactory_TypeaheadContainerComponent1);
        this._NgFor_3_6 = new import13.Wrapper_NgFor(this._appEl_3.vcRef, this._TemplateRef_3_5, this.parentInjector.get(import16.IterableDiffers), this.ref);
        this._text_4 = this.renderer.createText(this._el_1, '\n  ', null);
        this._text_5 = this.renderer.createText(parentRenderNode, '\n', null);
        var disposable_0 = this.renderer.listen(this._el_1, 'mouseleave', this.eventHandler(this._handle_mouseleave_1_0.bind(this)));
        this._map_0 = import4.pureProxy3(function (p0, p1, p2) {
            return {
                top: p0,
                left: p1,
                display: p2
            };
        });
        this.init([], [
            this._text_0,
            this._el_1,
            this._text_2,
            this._anchor_3,
            this._text_4,
            this._text_5
        ], [disposable_0], []);
        return null;
    };
    _View_TypeaheadContainerComponent0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import15.TemplateRef) && (3 === requestNodeIndex))) {
            return this._TemplateRef_3_5;
        }
        if (((token === import17.NgFor) && (3 === requestNodeIndex))) {
            return this._NgFor_3_6.context;
        }
        if (((token === import18.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 4)))) {
            return this._NgStyle_1_3.context;
        }
        return notFoundResult;
    };
    _View_TypeaheadContainerComponent0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_1 = this._map_0(this.context.top, this.context.left, 'block');
        this._NgStyle_1_3.check_ngStyle(currVal_1, throwOnChange, false);
        this._NgStyle_1_3.detectChangesInternal(this, this._el_1, throwOnChange);
        var currVal_2 = this.context.matches;
        this._NgFor_3_6.check_ngForOf(currVal_2, throwOnChange, false);
        this._NgFor_3_6.detectChangesInternal(this, this._anchor_3, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_TypeaheadContainerComponent0.prototype._handle_mouseleave_1_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.focusLost() !== false);
        return (true && pd_0);
    };
    return _View_TypeaheadContainerComponent0;
}(import1.AppView));
export function viewFactory_TypeaheadContainerComponent0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_TypeaheadContainerComponent === null)) {
        (renderType_TypeaheadContainerComponent = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, styles_TypeaheadContainerComponent, {}));
    }
    return new _View_TypeaheadContainerComponent0(viewUtils, parentInjector, declarationEl);
}
var _View_TypeaheadContainerComponent1 = (function (_super) {
    __extends(_View_TypeaheadContainerComponent1, _super);
    function _View_TypeaheadContainerComponent1(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_TypeaheadContainerComponent1, renderType_TypeaheadContainerComponent, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_TypeaheadContainerComponent1.prototype.createInternal = function (rootSelector) {
        this._text_0 = this.renderer.createText(null, '\n      ', null);
        this._anchor_1 = this.renderer.createTemplateAnchor(null, null);
        this._appEl_1 = new import3.AppElement(1, null, this, this._anchor_1);
        this._TemplateRef_1_5 = new import15.TemplateRef_(this._appEl_1, viewFactory_TypeaheadContainerComponent2);
        this._NgIf_1_6 = new import19.Wrapper_NgIf(this._appEl_1.vcRef, this._TemplateRef_1_5);
        this._text_2 = this.renderer.createText(null, '\n      ', null);
        this._anchor_3 = this.renderer.createTemplateAnchor(null, null);
        this._appEl_3 = new import3.AppElement(3, null, this, this._anchor_3);
        this._TemplateRef_3_5 = new import15.TemplateRef_(this._appEl_3, viewFactory_TypeaheadContainerComponent3);
        this._NgIf_3_6 = new import19.Wrapper_NgIf(this._appEl_3.vcRef, this._TemplateRef_3_5);
        this._text_4 = this.renderer.createText(null, '\n    ', null);
        this.init([].concat([
            this._text_0,
            this._appEl_1,
            this._text_2,
            this._appEl_3,
            this._text_4
        ]), [
            this._text_0,
            this._anchor_1,
            this._text_2,
            this._anchor_3,
            this._text_4
        ], [], []);
        return null;
    };
    _View_TypeaheadContainerComponent1.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import15.TemplateRef) && (1 === requestNodeIndex))) {
            return this._TemplateRef_1_5;
        }
        if (((token === import20.NgIf) && (1 === requestNodeIndex))) {
            return this._NgIf_1_6.context;
        }
        if (((token === import15.TemplateRef) && (3 === requestNodeIndex))) {
            return this._TemplateRef_3_5;
        }
        if (((token === import20.NgIf) && (3 === requestNodeIndex))) {
            return this._NgIf_3_6.context;
        }
        return notFoundResult;
    };
    _View_TypeaheadContainerComponent1.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = this.context.$implicit.isHeader();
        this._NgIf_1_6.check_ngIf(currVal_0, throwOnChange, false);
        this._NgIf_1_6.detectChangesInternal(this, this._anchor_1, throwOnChange);
        var currVal_1 = !this.context.$implicit.isHeader();
        this._NgIf_3_6.check_ngIf(currVal_1, throwOnChange, false);
        this._NgIf_3_6.detectChangesInternal(this, this._anchor_3, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_TypeaheadContainerComponent1;
}(import1.AppView));
function viewFactory_TypeaheadContainerComponent1(viewUtils, parentInjector, declarationEl) {
    return new _View_TypeaheadContainerComponent1(viewUtils, parentInjector, declarationEl);
}
var _View_TypeaheadContainerComponent2 = (function (_super) {
    __extends(_View_TypeaheadContainerComponent2, _super);
    function _View_TypeaheadContainerComponent2(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_TypeaheadContainerComponent2, renderType_TypeaheadContainerComponent, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_TypeaheadContainerComponent2.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'li', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'dropdown-header');
        this._text_1 = this.renderer.createText(this._el_0, '', null);
        this._expr_0 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1
        ], [], []);
        return null;
    };
    _View_TypeaheadContainerComponent2.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = import4.interpolate(1, '', this.parent.context.$implicit, '');
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setText(this._text_1, currVal_0);
            this._expr_0 = currVal_0;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_TypeaheadContainerComponent2;
}(import1.AppView));
function viewFactory_TypeaheadContainerComponent2(viewUtils, parentInjector, declarationEl) {
    return new _View_TypeaheadContainerComponent2(viewUtils, parentInjector, declarationEl);
}
var _View_TypeaheadContainerComponent3 = (function (_super) {
    __extends(_View_TypeaheadContainerComponent3, _super);
    function _View_TypeaheadContainerComponent3(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_TypeaheadContainerComponent3, renderType_TypeaheadContainerComponent, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_TypeaheadContainerComponent3.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'li', null);
        this._text_1 = this.renderer.createText(this._el_0, '\n        ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_2 = new import3.AppElement(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import15.TemplateRef_(this._appEl_2, viewFactory_TypeaheadContainerComponent4);
        this._NgIf_2_6 = new import19.Wrapper_NgIf(this._appEl_2.vcRef, this._TemplateRef_2_5);
        this._text_3 = this.renderer.createText(this._el_0, '\n        ', null);
        this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_4 = new import3.AppElement(4, 0, this, this._anchor_4);
        this._TemplateRef_4_5 = new import15.TemplateRef_(this._appEl_4, viewFactory_TypeaheadContainerComponent5);
        this._NgIf_4_6 = new import19.Wrapper_NgIf(this._appEl_4.vcRef, this._TemplateRef_4_5);
        this._text_5 = this.renderer.createText(this._el_0, '\n      ', null);
        this._expr_1 = import7.UNINITIALIZED;
        var disposable_0 = this.renderer.listen(this._el_0, 'mouseenter', this.eventHandler(this._handle_mouseenter_0_0.bind(this)));
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3,
            this._anchor_4,
            this._text_5
        ], [disposable_0], []);
        return null;
    };
    _View_TypeaheadContainerComponent3.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import15.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import20.NgIf) && (2 === requestNodeIndex))) {
            return this._NgIf_2_6.context;
        }
        if (((token === import15.TemplateRef) && (4 === requestNodeIndex))) {
            return this._TemplateRef_4_5;
        }
        if (((token === import20.NgIf) && (4 === requestNodeIndex))) {
            return this._NgIf_4_6.context;
        }
        return notFoundResult;
    };
    _View_TypeaheadContainerComponent3.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_2 = !this.parent.parent.context.itemTemplate;
        this._NgIf_2_6.check_ngIf(currVal_2, throwOnChange, false);
        this._NgIf_2_6.detectChangesInternal(this, this._anchor_2, throwOnChange);
        var currVal_3 = this.parent.parent.context.itemTemplate;
        this._NgIf_4_6.check_ngIf(currVal_3, throwOnChange, false);
        this._NgIf_4_6.detectChangesInternal(this, this._anchor_4, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_1 = this.parent.parent.context.isActive(this.parent.context.$implicit);
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementClass(this._el_0, 'active', currVal_1);
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_TypeaheadContainerComponent3.prototype._handle_mouseenter_0_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.parent.parent.context.selectActive(this.parent.context.$implicit) !== false);
        return (true && pd_0);
    };
    return _View_TypeaheadContainerComponent3;
}(import1.AppView));
function viewFactory_TypeaheadContainerComponent3(viewUtils, parentInjector, declarationEl) {
    return new _View_TypeaheadContainerComponent3(viewUtils, parentInjector, declarationEl);
}
var _View_TypeaheadContainerComponent4 = (function (_super) {
    __extends(_View_TypeaheadContainerComponent4, _super);
    function _View_TypeaheadContainerComponent4(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_TypeaheadContainerComponent4, renderType_TypeaheadContainerComponent, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_TypeaheadContainerComponent4.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'a', null);
        this.renderer.setElementAttribute(this._el_0, 'href', '#');
        this.renderer.setElementAttribute(this._el_0, 'tabindex', '-1');
        this._expr_1 = import7.UNINITIALIZED;
        var disposable_0 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_0.bind(this)));
        this.init([].concat([this._el_0]), [this._el_0], [disposable_0], []);
        return null;
    };
    _View_TypeaheadContainerComponent4.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_1 = this.parent.parent.parent.context.hightlight(this.parent.parent.context.$implicit, this.parent.parent.parent.context.query);
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementProperty(this._el_0, 'innerHTML', this.viewUtils.sanitizer.sanitize(import21.SecurityContext.HTML, currVal_1));
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_TypeaheadContainerComponent4.prototype._handle_click_0_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.parent.parent.parent.context.selectMatch(this.parent.parent.context.$implicit, $event) !== false);
        return (true && pd_0);
    };
    return _View_TypeaheadContainerComponent4;
}(import1.AppView));
function viewFactory_TypeaheadContainerComponent4(viewUtils, parentInjector, declarationEl) {
    return new _View_TypeaheadContainerComponent4(viewUtils, parentInjector, declarationEl);
}
var _View_TypeaheadContainerComponent5 = (function (_super) {
    __extends(_View_TypeaheadContainerComponent5, _super);
    function _View_TypeaheadContainerComponent5(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_TypeaheadContainerComponent5, renderType_TypeaheadContainerComponent, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_TypeaheadContainerComponent5.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.renderer.createElement(null, 'a', null);
        this.renderer.setElementAttribute(this._el_0, 'href', '#');
        this.renderer.setElementAttribute(this._el_0, 'tabindex', '-1');
        this._text_1 = this.renderer.createText(this._el_0, '\n            ', null);
        this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0, null);
        this._appEl_2 = new import3.AppElement(2, 0, this, this._anchor_2);
        this._TemplateRef_2_5 = new import15.TemplateRef_(this._appEl_2, viewFactory_TypeaheadContainerComponent6);
        this._NgTemplateOutlet_2_6 = new import22.Wrapper_NgTemplateOutlet(this._appEl_2.vcRef);
        this._text_3 = this.renderer.createText(this._el_0, '\n        ', null);
        var disposable_0 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_0.bind(this)));
        this._map_0 = import4.pureProxy2(function (p0, p1) {
            return {
                item: p0,
                index: p1
            };
        });
        this.init([].concat([this._el_0]), [
            this._el_0,
            this._text_1,
            this._anchor_2,
            this._text_3
        ], [disposable_0], []);
        return null;
    };
    _View_TypeaheadContainerComponent5.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import15.TemplateRef) && (2 === requestNodeIndex))) {
            return this._TemplateRef_2_5;
        }
        if (((token === import23.NgTemplateOutlet) && (2 === requestNodeIndex))) {
            return this._NgTemplateOutlet_2_6.context;
        }
        return notFoundResult;
    };
    _View_TypeaheadContainerComponent5.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_1 = this._map_0(this.parent.parent.context.$implicit.item, this.parent.parent.context.index);
        this._NgTemplateOutlet_2_6.check_ngOutletContext(currVal_1, throwOnChange, false);
        var currVal_2 = this.parent.parent.parent.context.itemTemplate;
        this._NgTemplateOutlet_2_6.check_ngTemplateOutlet(currVal_2, throwOnChange, false);
        this._NgTemplateOutlet_2_6.detectChangesInternal(this, this._anchor_2, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_TypeaheadContainerComponent5.prototype._handle_click_0_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.parent.parent.parent.context.selectMatch(this.parent.parent.context.$implicit, $event) !== false);
        return (true && pd_0);
    };
    return _View_TypeaheadContainerComponent5;
}(import1.AppView));
function viewFactory_TypeaheadContainerComponent5(viewUtils, parentInjector, declarationEl) {
    return new _View_TypeaheadContainerComponent5(viewUtils, parentInjector, declarationEl);
}
var _View_TypeaheadContainerComponent6 = (function (_super) {
    __extends(_View_TypeaheadContainerComponent6, _super);
    function _View_TypeaheadContainerComponent6(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_TypeaheadContainerComponent6, renderType_TypeaheadContainerComponent, import6.ViewType.EMBEDDED, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_TypeaheadContainerComponent6.prototype.createInternal = function (rootSelector) {
        this._text_0 = this.renderer.createText(null, '\n            ', null);
        this.init([].concat([this._text_0]), [this._text_0], [], []);
        return null;
    };
    return _View_TypeaheadContainerComponent6;
}(import1.AppView));
function viewFactory_TypeaheadContainerComponent6(viewUtils, parentInjector, declarationEl) {
    return new _View_TypeaheadContainerComponent6(viewUtils, parentInjector, declarationEl);
}
