// ng2-bootstrap/components/timepicker/timepicker.component.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/timepicker/timepicker.component';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
import * as import4 from '@angular/core/src/render/api';
import * as import5 from '@angular/core/src/linker/element';
import * as import6 from '@angular/forms/src/directives/ng_model';
import * as import7 from '@angular/core/src/di/injector';
import * as import8 from '@angular/core/src/linker/view_type';
import * as import9 from '@angular/forms/src/directives/control_container';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/component_factory';
import * as import12 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import13 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import14 from '../../../@angular/forms/src/directives/default_value_accessor.ngfactory';
import * as import15 from '../../../@angular/forms/src/directives/validators.ngfactory';
import * as import16 from '../../../@angular/forms/src/directives/ng_model.ngfactory';
import * as import17 from '../../../@angular/forms/src/directives/ng_control_status.ngfactory';
import * as import18 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import19 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import20 from '@angular/core/src/linker/element_ref';
import * as import21 from '@angular/core/src/linker/template_ref';
import * as import22 from '@angular/common/src/directives/ng_class';
import * as import23 from '@angular/common/src/directives/ng_if';
import * as import24 from '@angular/forms/src/directives/default_value_accessor';
import * as import25 from '@angular/forms/src/directives/validators';
import * as import26 from '@angular/forms/src/validators';
import * as import27 from '@angular/forms/src/directives/control_value_accessor';
import * as import28 from '@angular/forms/src/directives/ng_control';
import * as import29 from '@angular/forms/src/directives/ng_control_status';
export class Wrapper_TimepickerComponent {
  context:import0.TimepickerComponent;
  changed:boolean;
  /*private*/ _hourStep:any;
  /*private*/ _minuteStep:any;
  /*private*/ _readonlyInput:any;
  /*private*/ _mousewheel:any;
  /*private*/ _arrowkeys:any;
  /*private*/ _showSpinners:any;
  /*private*/ _min:any;
  /*private*/ _max:any;
  /*private*/ _meridians:any;
  /*private*/ _showMeridian:any;
  constructor(p0:any) {
    this.changed = false;
    this._hourStep = import1.UNINITIALIZED;
    this._minuteStep = import1.UNINITIALIZED;
    this._readonlyInput = import1.UNINITIALIZED;
    this._mousewheel = import1.UNINITIALIZED;
    this._arrowkeys = import1.UNINITIALIZED;
    this._showSpinners = import1.UNINITIALIZED;
    this._min = import1.UNINITIALIZED;
    this._max = import1.UNINITIALIZED;
    this._meridians = import1.UNINITIALIZED;
    this._showMeridian = import1.UNINITIALIZED;
    this.context = new import0.TimepickerComponent(p0);
  }
  check_hourStep(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._hourStep,currValue))) {
      this.changed = true;
      this.context.hourStep = currValue;
      this._hourStep = currValue;
    }
  }
  check_minuteStep(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._minuteStep,currValue))) {
      this.changed = true;
      this.context.minuteStep = currValue;
      this._minuteStep = currValue;
    }
  }
  check_readonlyInput(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._readonlyInput,currValue))) {
      this.changed = true;
      this.context.readonlyInput = currValue;
      this._readonlyInput = currValue;
    }
  }
  check_mousewheel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._mousewheel,currValue))) {
      this.changed = true;
      this.context.mousewheel = currValue;
      this._mousewheel = currValue;
    }
  }
  check_arrowkeys(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._arrowkeys,currValue))) {
      this.changed = true;
      this.context.arrowkeys = currValue;
      this._arrowkeys = currValue;
    }
  }
  check_showSpinners(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._showSpinners,currValue))) {
      this.changed = true;
      this.context.showSpinners = currValue;
      this._showSpinners = currValue;
    }
  }
  check_min(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._min,currValue))) {
      this.changed = true;
      this.context.min = currValue;
      this._min = currValue;
    }
  }
  check_max(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._max,currValue))) {
      this.changed = true;
      this.context.max = currValue;
      this._max = currValue;
    }
  }
  check_meridians(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._meridians,currValue))) {
      this.changed = true;
      this.context.meridians = currValue;
      this._meridians = currValue;
    }
  }
  check_showMeridian(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import2.checkBinding(throwOnChange,this._showMeridian,currValue))) {
      this.changed = true;
      this.context.showMeridian = currValue;
      this._showMeridian = currValue;
    }
  }
  detectChangesInternal(view:import3.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
}
var renderType_TimepickerComponent_Host:import4.RenderComponentType = (null as any);
class _View_TimepickerComponent_Host0 extends import3.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import5.AppElement;
  _NgModel_0_4:import6.NgModel;
  _TimepickerComponent_0_5:Wrapper_TimepickerComponent;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_TimepickerComponent_Host0,renderType_TimepickerComponent_Host,import8.ViewType.HOST,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.selectOrCreateHostElement('timepicker',rootSelector,(null as any));
    this.renderer.setElementAttribute(this._el_0,'ngModel','');
    this._appEl_0 = new import5.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_TimepickerComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._NgModel_0_4 = new import6.NgModel(this.parentInjector.get(import9.ControlContainer,(null as any)),(null as any),(null as any),(null as any));
    this._TimepickerComponent_0_5 = new Wrapper_TimepickerComponent(this._NgModel_0_4);
    this._appEl_0.initComponent(this._TimepickerComponent_0_5.context,([] as any[]),compView_0);
    compView_0.create(this._TimepickerComponent_0_5.context,this.projectableNodes,(null as any));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import6.NgModel) && (0 === requestNodeIndex))) { return this._NgModel_0_4; }
    if (((token === import0.TimepickerComponent) && (0 === requestNodeIndex))) { return this._TimepickerComponent_0_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._TimepickerComponent_0_5.detectChangesInternal(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._NgModel_0_4.ngOnDestroy();
  }
}
function viewFactory_TimepickerComponent_Host0(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  if ((renderType_TimepickerComponent_Host === (null as any))) { (renderType_TimepickerComponent_Host = viewUtils.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_TimepickerComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const TimepickerComponentNgFactory:import11.ComponentFactory<import0.TimepickerComponent> = new import11.ComponentFactory<import0.TimepickerComponent>('timepicker[ngModel]',viewFactory_TimepickerComponent_Host0,import0.TimepickerComponent);
const styles_TimepickerComponent:any[] = ([] as any[]);
var renderType_TimepickerComponent:import4.RenderComponentType = (null as any);
class _View_TimepickerComponent0 extends import3.AppView<import0.TimepickerComponent> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _text_4:any;
  _el_5:any;
  _NgClass_5_3:import12.Wrapper_NgClass;
  _text_6:any;
  _el_7:any;
  _el_8:any;
  _NgClass_8_3:import12.Wrapper_NgClass;
  _el_9:any;
  _text_10:any;
  _el_11:any;
  _text_12:any;
  _text_13:any;
  _el_14:any;
  _el_15:any;
  _NgClass_15_3:import12.Wrapper_NgClass;
  _el_16:any;
  _text_17:any;
  _anchor_18:any;
  /*private*/ _appEl_18:import5.AppElement;
  _TemplateRef_18_5:any;
  _NgIf_18_6:import13.Wrapper_NgIf;
  _text_19:any;
  _text_20:any;
  _el_21:any;
  _text_22:any;
  _el_23:any;
  _NgClass_23_3:import12.Wrapper_NgClass;
  _text_24:any;
  _el_25:any;
  _DefaultValueAccessor_25_3:import14.Wrapper_DefaultValueAccessor;
  _MaxLengthValidator_25_4:import15.Wrapper_MaxLengthValidator;
  _NG_VALIDATORS_25_5:any[];
  _NG_VALUE_ACCESSOR_25_6:any[];
  _NgModel_25_7:import16.Wrapper_NgModel;
  _NgControl_25_8:any;
  _NgControlStatus_25_9:import17.Wrapper_NgControlStatus;
  _text_26:any;
  _text_27:any;
  _el_28:any;
  _text_29:any;
  _text_30:any;
  _el_31:any;
  _NgClass_31_3:import12.Wrapper_NgClass;
  _text_32:any;
  _el_33:any;
  _DefaultValueAccessor_33_3:import14.Wrapper_DefaultValueAccessor;
  _MaxLengthValidator_33_4:import15.Wrapper_MaxLengthValidator;
  _NG_VALIDATORS_33_5:any[];
  _NG_VALUE_ACCESSOR_33_6:any[];
  _NgModel_33_7:import16.Wrapper_NgModel;
  _NgControl_33_8:any;
  _NgControlStatus_33_9:import17.Wrapper_NgControlStatus;
  _text_34:any;
  _text_35:any;
  _anchor_36:any;
  /*private*/ _appEl_36:import5.AppElement;
  _TemplateRef_36_5:any;
  _NgIf_36_6:import13.Wrapper_NgIf;
  _text_37:any;
  _text_38:any;
  _el_39:any;
  _NgClass_39_3:import12.Wrapper_NgClass;
  _text_40:any;
  _el_41:any;
  _el_42:any;
  _NgClass_42_3:import12.Wrapper_NgClass;
  _el_43:any;
  _text_44:any;
  _el_45:any;
  _text_46:any;
  _text_47:any;
  _el_48:any;
  _el_49:any;
  _NgClass_49_3:import12.Wrapper_NgClass;
  _el_50:any;
  _text_51:any;
  _anchor_52:any;
  /*private*/ _appEl_52:import5.AppElement;
  _TemplateRef_52_5:any;
  _NgIf_52_6:import13.Wrapper_NgIf;
  _text_53:any;
  _text_54:any;
  _text_55:any;
  _text_56:any;
  _map_0:any;
  _map_1:any;
  _map_2:any;
  _map_3:any;
  /*private*/ _expr_16:any;
  /*private*/ _expr_18:any;
  /*private*/ _expr_20:any;
  /*private*/ _expr_21:any;
  /*private*/ _expr_22:any;
  /*private*/ _expr_23:any;
  /*private*/ _expr_24:any;
  /*private*/ _expr_25:any;
  _map_4:any;
  /*private*/ _expr_33:any;
  /*private*/ _expr_35:any;
  /*private*/ _expr_37:any;
  /*private*/ _expr_38:any;
  /*private*/ _expr_39:any;
  /*private*/ _expr_40:any;
  /*private*/ _expr_41:any;
  /*private*/ _expr_42:any;
  _map_5:any;
  _map_6:any;
  _map_7:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_TimepickerComponent0,renderType_TimepickerComponent,import8.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this._el_1 = this.renderer.createElement(parentRenderNode,'table',(null as any));
    this._text_2 = this.renderer.createText(this._el_1,'\n      ',(null as any));
    this._el_3 = this.renderer.createElement(this._el_1,'tbody',(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'\n        ',(null as any));
    this._el_5 = this.renderer.createElement(this._el_3,'tr',(null as any));
    this.renderer.setElementAttribute(this._el_5,'class','text-center');
    this._NgClass_5_3 = new import12.Wrapper_NgClass(this.parentInjector.get(import18.IterableDiffers),this.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_5),this.renderer);
    this._text_6 = this.renderer.createText(this._el_5,'\n          ',(null as any));
    this._el_7 = this.renderer.createElement(this._el_5,'td',(null as any));
    this._el_8 = this.renderer.createElement(this._el_7,'a',(null as any));
    this.renderer.setElementAttribute(this._el_8,'class','btn btn-link');
    this._NgClass_8_3 = new import12.Wrapper_NgClass(this.parentInjector.get(import18.IterableDiffers),this.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_8),this.renderer);
    this._el_9 = this.renderer.createElement(this._el_8,'span',(null as any));
    this.renderer.setElementAttribute(this._el_9,'class','glyphicon glyphicon-chevron-up');
    this._text_10 = this.renderer.createText(this._el_5,'\n          ',(null as any));
    this._el_11 = this.renderer.createElement(this._el_5,'td',(null as any));
    this._text_12 = this.renderer.createText(this._el_11,' ',(null as any));
    this._text_13 = this.renderer.createText(this._el_5,'\n          ',(null as any));
    this._el_14 = this.renderer.createElement(this._el_5,'td',(null as any));
    this._el_15 = this.renderer.createElement(this._el_14,'a',(null as any));
    this.renderer.setElementAttribute(this._el_15,'class','btn btn-link');
    this._NgClass_15_3 = new import12.Wrapper_NgClass(this.parentInjector.get(import18.IterableDiffers),this.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_15),this.renderer);
    this._el_16 = this.renderer.createElement(this._el_15,'span',(null as any));
    this.renderer.setElementAttribute(this._el_16,'class','glyphicon glyphicon-chevron-up');
    this._text_17 = this.renderer.createText(this._el_5,'\n          ',(null as any));
    this._anchor_18 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._appEl_18 = new import5.AppElement(18,5,this,this._anchor_18);
    this._TemplateRef_18_5 = new import21.TemplateRef_(this._appEl_18,viewFactory_TimepickerComponent1);
    this._NgIf_18_6 = new import13.Wrapper_NgIf(this._appEl_18.vcRef,this._TemplateRef_18_5);
    this._text_19 = this.renderer.createText(this._el_5,'\n        ',(null as any));
    this._text_20 = this.renderer.createText(this._el_3,'\n        ',(null as any));
    this._el_21 = this.renderer.createElement(this._el_3,'tr',(null as any));
    this._text_22 = this.renderer.createText(this._el_21,'\n          ',(null as any));
    this._el_23 = this.renderer.createElement(this._el_21,'td',(null as any));
    this.renderer.setElementAttribute(this._el_23,'class','form-group');
    this._NgClass_23_3 = new import12.Wrapper_NgClass(this.parentInjector.get(import18.IterableDiffers),this.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_23),this.renderer);
    this._text_24 = this.renderer.createText(this._el_23,'\n            ',(null as any));
    this._el_25 = this.renderer.createElement(this._el_23,'input',(null as any));
    this.renderer.setElementAttribute(this._el_25,'class','form-control text-center');
    this.renderer.setElementAttribute(this._el_25,'maxlength','2');
    this.renderer.setElementAttribute(this._el_25,'style','width:50px;');
    this.renderer.setElementAttribute(this._el_25,'type','text');
    this._DefaultValueAccessor_25_3 = new import14.Wrapper_DefaultValueAccessor(this.renderer,new import20.ElementRef(this._el_25));
    this._MaxLengthValidator_25_4 = new import15.Wrapper_MaxLengthValidator();
    this._NG_VALIDATORS_25_5 = [this._MaxLengthValidator_25_4.context];
    this._NG_VALUE_ACCESSOR_25_6 = [this._DefaultValueAccessor_25_3.context];
    this._NgModel_25_7 = new import16.Wrapper_NgModel((null as any),this._NG_VALIDATORS_25_5,(null as any),this._NG_VALUE_ACCESSOR_25_6);
    this._NgControl_25_8 = this._NgModel_25_7.context;
    this._NgControlStatus_25_9 = new import17.Wrapper_NgControlStatus(this._NgControl_25_8);
    this._text_26 = this.renderer.createText(this._el_23,'\n          ',(null as any));
    this._text_27 = this.renderer.createText(this._el_21,'\n          ',(null as any));
    this._el_28 = this.renderer.createElement(this._el_21,'td',(null as any));
    this._text_29 = this.renderer.createText(this._el_28,':',(null as any));
    this._text_30 = this.renderer.createText(this._el_21,'\n          ',(null as any));
    this._el_31 = this.renderer.createElement(this._el_21,'td',(null as any));
    this.renderer.setElementAttribute(this._el_31,'class','form-group');
    this._NgClass_31_3 = new import12.Wrapper_NgClass(this.parentInjector.get(import18.IterableDiffers),this.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_31),this.renderer);
    this._text_32 = this.renderer.createText(this._el_31,'\n            ',(null as any));
    this._el_33 = this.renderer.createElement(this._el_31,'input',(null as any));
    this.renderer.setElementAttribute(this._el_33,'class','form-control text-center');
    this.renderer.setElementAttribute(this._el_33,'maxlength','2');
    this.renderer.setElementAttribute(this._el_33,'style','width:50px;');
    this.renderer.setElementAttribute(this._el_33,'type','text');
    this._DefaultValueAccessor_33_3 = new import14.Wrapper_DefaultValueAccessor(this.renderer,new import20.ElementRef(this._el_33));
    this._MaxLengthValidator_33_4 = new import15.Wrapper_MaxLengthValidator();
    this._NG_VALIDATORS_33_5 = [this._MaxLengthValidator_33_4.context];
    this._NG_VALUE_ACCESSOR_33_6 = [this._DefaultValueAccessor_33_3.context];
    this._NgModel_33_7 = new import16.Wrapper_NgModel((null as any),this._NG_VALIDATORS_33_5,(null as any),this._NG_VALUE_ACCESSOR_33_6);
    this._NgControl_33_8 = this._NgModel_33_7.context;
    this._NgControlStatus_33_9 = new import17.Wrapper_NgControlStatus(this._NgControl_33_8);
    this._text_34 = this.renderer.createText(this._el_31,'\n          ',(null as any));
    this._text_35 = this.renderer.createText(this._el_21,'\n          ',(null as any));
    this._anchor_36 = this.renderer.createTemplateAnchor(this._el_21,(null as any));
    this._appEl_36 = new import5.AppElement(36,21,this,this._anchor_36);
    this._TemplateRef_36_5 = new import21.TemplateRef_(this._appEl_36,viewFactory_TimepickerComponent2);
    this._NgIf_36_6 = new import13.Wrapper_NgIf(this._appEl_36.vcRef,this._TemplateRef_36_5);
    this._text_37 = this.renderer.createText(this._el_21,'\n        ',(null as any));
    this._text_38 = this.renderer.createText(this._el_3,'\n        ',(null as any));
    this._el_39 = this.renderer.createElement(this._el_3,'tr',(null as any));
    this.renderer.setElementAttribute(this._el_39,'class','text-center');
    this._NgClass_39_3 = new import12.Wrapper_NgClass(this.parentInjector.get(import18.IterableDiffers),this.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_39),this.renderer);
    this._text_40 = this.renderer.createText(this._el_39,'\n          ',(null as any));
    this._el_41 = this.renderer.createElement(this._el_39,'td',(null as any));
    this._el_42 = this.renderer.createElement(this._el_41,'a',(null as any));
    this.renderer.setElementAttribute(this._el_42,'class','btn btn-link');
    this._NgClass_42_3 = new import12.Wrapper_NgClass(this.parentInjector.get(import18.IterableDiffers),this.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_42),this.renderer);
    this._el_43 = this.renderer.createElement(this._el_42,'span',(null as any));
    this.renderer.setElementAttribute(this._el_43,'class','glyphicon glyphicon-chevron-down');
    this._text_44 = this.renderer.createText(this._el_39,'\n          ',(null as any));
    this._el_45 = this.renderer.createElement(this._el_39,'td',(null as any));
    this._text_46 = this.renderer.createText(this._el_45,' ',(null as any));
    this._text_47 = this.renderer.createText(this._el_39,'\n          ',(null as any));
    this._el_48 = this.renderer.createElement(this._el_39,'td',(null as any));
    this._el_49 = this.renderer.createElement(this._el_48,'a',(null as any));
    this.renderer.setElementAttribute(this._el_49,'class','btn btn-link');
    this._NgClass_49_3 = new import12.Wrapper_NgClass(this.parentInjector.get(import18.IterableDiffers),this.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_49),this.renderer);
    this._el_50 = this.renderer.createElement(this._el_49,'span',(null as any));
    this.renderer.setElementAttribute(this._el_50,'class','glyphicon glyphicon-chevron-down');
    this._text_51 = this.renderer.createText(this._el_39,'\n          ',(null as any));
    this._anchor_52 = this.renderer.createTemplateAnchor(this._el_39,(null as any));
    this._appEl_52 = new import5.AppElement(52,39,this,this._anchor_52);
    this._TemplateRef_52_5 = new import21.TemplateRef_(this._appEl_52,viewFactory_TimepickerComponent3);
    this._NgIf_52_6 = new import13.Wrapper_NgIf(this._appEl_52.vcRef,this._TemplateRef_52_5);
    this._text_53 = this.renderer.createText(this._el_39,'\n        ',(null as any));
    this._text_54 = this.renderer.createText(this._el_3,'\n      ',(null as any));
    this._text_55 = this.renderer.createText(this._el_1,'\n    ',(null as any));
    this._text_56 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this._map_0 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
    var disposable_0:Function = this.renderer.listen(this._el_8,'click',this.eventHandler(this._handle_click_8_0.bind(this)));
    this._map_1 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    var disposable_1:Function = this.renderer.listen(this._el_15,'click',this.eventHandler(this._handle_click_15_0.bind(this)));
    this._map_2 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this._map_3 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {'has-error': p0};
    });
    this._expr_16 = import1.UNINITIALIZED;
    var disposable_2:Function = this.renderer.listen(this._el_25,'ngModelChange',this.eventHandler(this._handle_ngModelChange_25_0.bind(this)));
    var disposable_3:Function = this.renderer.listen(this._el_25,'change',this.eventHandler(this._handle_change_25_1.bind(this)));
    var disposable_4:Function = this.renderer.listen(this._el_25,'blur',this.eventHandler(this._handle_blur_25_2.bind(this)));
    var disposable_5:Function = this.renderer.listen(this._el_25,'input',this.eventHandler(this._handle_input_25_3.bind(this)));
    this._expr_18 = import1.UNINITIALIZED;
    const subscription_0:any = this._NgModel_25_7.context.update.subscribe(this.eventHandler(this._handle_ngModelChange_25_0.bind(this)));
    this._expr_20 = import1.UNINITIALIZED;
    this._expr_21 = import1.UNINITIALIZED;
    this._expr_22 = import1.UNINITIALIZED;
    this._expr_23 = import1.UNINITIALIZED;
    this._expr_24 = import1.UNINITIALIZED;
    this._expr_25 = import1.UNINITIALIZED;
    this._map_4 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {'has-error': p0};
    });
    this._expr_33 = import1.UNINITIALIZED;
    var disposable_6:Function = this.renderer.listen(this._el_33,'ngModelChange',this.eventHandler(this._handle_ngModelChange_33_0.bind(this)));
    var disposable_7:Function = this.renderer.listen(this._el_33,'change',this.eventHandler(this._handle_change_33_1.bind(this)));
    var disposable_8:Function = this.renderer.listen(this._el_33,'blur',this.eventHandler(this._handle_blur_33_2.bind(this)));
    var disposable_9:Function = this.renderer.listen(this._el_33,'input',this.eventHandler(this._handle_input_33_3.bind(this)));
    this._expr_35 = import1.UNINITIALIZED;
    const subscription_1:any = this._NgModel_33_7.context.update.subscribe(this.eventHandler(this._handle_ngModelChange_33_0.bind(this)));
    this._expr_37 = import1.UNINITIALIZED;
    this._expr_38 = import1.UNINITIALIZED;
    this._expr_39 = import1.UNINITIALIZED;
    this._expr_40 = import1.UNINITIALIZED;
    this._expr_41 = import1.UNINITIALIZED;
    this._expr_42 = import1.UNINITIALIZED;
    this._map_5 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
    var disposable_10:Function = this.renderer.listen(this._el_42,'click',this.eventHandler(this._handle_click_42_0.bind(this)));
    this._map_6 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    var disposable_11:Function = this.renderer.listen(this._el_49,'click',this.eventHandler(this._handle_click_49_0.bind(this)));
    this._map_7 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this.init(([] as any[]),[
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._el_8,
      this._el_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._text_13,
      this._el_14,
      this._el_15,
      this._el_16,
      this._text_17,
      this._anchor_18,
      this._text_19,
      this._text_20,
      this._el_21,
      this._text_22,
      this._el_23,
      this._text_24,
      this._el_25,
      this._text_26,
      this._text_27,
      this._el_28,
      this._text_29,
      this._text_30,
      this._el_31,
      this._text_32,
      this._el_33,
      this._text_34,
      this._text_35,
      this._anchor_36,
      this._text_37,
      this._text_38,
      this._el_39,
      this._text_40,
      this._el_41,
      this._el_42,
      this._el_43,
      this._text_44,
      this._el_45,
      this._text_46,
      this._text_47,
      this._el_48,
      this._el_49,
      this._el_50,
      this._text_51,
      this._anchor_52,
      this._text_53,
      this._text_54,
      this._text_55,
      this._text_56
    ]
    ,[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3,
      disposable_4,
      disposable_5,
      disposable_6,
      disposable_7,
      disposable_8,
      disposable_9,
      disposable_10,
      disposable_11
    ]
    ,[
      subscription_0,
      subscription_1
    ]
    );
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && ((8 <= requestNodeIndex) && (requestNodeIndex <= 9)))) { return this._NgClass_8_3.context; }
    if (((token === import22.NgClass) && ((15 <= requestNodeIndex) && (requestNodeIndex <= 16)))) { return this._NgClass_15_3.context; }
    if (((token === import21.TemplateRef) && (18 === requestNodeIndex))) { return this._TemplateRef_18_5; }
    if (((token === import23.NgIf) && (18 === requestNodeIndex))) { return this._NgIf_18_6.context; }
    if (((token === import22.NgClass) && ((5 <= requestNodeIndex) && (requestNodeIndex <= 19)))) { return this._NgClass_5_3.context; }
    if (((token === import24.DefaultValueAccessor) && (25 === requestNodeIndex))) { return this._DefaultValueAccessor_25_3.context; }
    if (((token === import25.MaxLengthValidator) && (25 === requestNodeIndex))) { return this._MaxLengthValidator_25_4.context; }
    if (((token === import26.NG_VALIDATORS) && (25 === requestNodeIndex))) { return this._NG_VALIDATORS_25_5; }
    if (((token === import27.NG_VALUE_ACCESSOR) && (25 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_25_6; }
    if (((token === import6.NgModel) && (25 === requestNodeIndex))) { return this._NgModel_25_7.context; }
    if (((token === import28.NgControl) && (25 === requestNodeIndex))) { return this._NgControl_25_8; }
    if (((token === import29.NgControlStatus) && (25 === requestNodeIndex))) { return this._NgControlStatus_25_9.context; }
    if (((token === import22.NgClass) && ((23 <= requestNodeIndex) && (requestNodeIndex <= 26)))) { return this._NgClass_23_3.context; }
    if (((token === import24.DefaultValueAccessor) && (33 === requestNodeIndex))) { return this._DefaultValueAccessor_33_3.context; }
    if (((token === import25.MaxLengthValidator) && (33 === requestNodeIndex))) { return this._MaxLengthValidator_33_4.context; }
    if (((token === import26.NG_VALIDATORS) && (33 === requestNodeIndex))) { return this._NG_VALIDATORS_33_5; }
    if (((token === import27.NG_VALUE_ACCESSOR) && (33 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_33_6; }
    if (((token === import6.NgModel) && (33 === requestNodeIndex))) { return this._NgModel_33_7.context; }
    if (((token === import28.NgControl) && (33 === requestNodeIndex))) { return this._NgControl_33_8; }
    if (((token === import29.NgControlStatus) && (33 === requestNodeIndex))) { return this._NgControlStatus_33_9.context; }
    if (((token === import22.NgClass) && ((31 <= requestNodeIndex) && (requestNodeIndex <= 34)))) { return this._NgClass_31_3.context; }
    if (((token === import21.TemplateRef) && (36 === requestNodeIndex))) { return this._TemplateRef_36_5; }
    if (((token === import23.NgIf) && (36 === requestNodeIndex))) { return this._NgIf_36_6.context; }
    if (((token === import22.NgClass) && ((42 <= requestNodeIndex) && (requestNodeIndex <= 43)))) { return this._NgClass_42_3.context; }
    if (((token === import22.NgClass) && ((49 <= requestNodeIndex) && (requestNodeIndex <= 50)))) { return this._NgClass_49_3.context; }
    if (((token === import21.TemplateRef) && (52 === requestNodeIndex))) { return this._TemplateRef_52_5; }
    if (((token === import23.NgIf) && (52 === requestNodeIndex))) { return this._NgIf_52_6.context; }
    if (((token === import22.NgClass) && ((39 <= requestNodeIndex) && (requestNodeIndex <= 53)))) { return this._NgClass_39_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0:any = 'text-center';
    this._NgClass_5_3.check_klass(currVal_0,throwOnChange,false);
    const currVal_1:any = this._map_0((!this.context.showSpinners || this.context.readonlyInput));
    this._NgClass_5_3.check_ngClass(currVal_1,throwOnChange,false);
    this._NgClass_5_3.detectChangesInternal(this,this._el_5,throwOnChange);
    const currVal_3:any = 'btn btn-link';
    this._NgClass_8_3.check_klass(currVal_3,throwOnChange,false);
    const currVal_4:any = this._map_1(this.context.noIncrementHours());
    this._NgClass_8_3.check_ngClass(currVal_4,throwOnChange,false);
    this._NgClass_8_3.detectChangesInternal(this,this._el_8,throwOnChange);
    const currVal_6:any = 'btn btn-link';
    this._NgClass_15_3.check_klass(currVal_6,throwOnChange,false);
    const currVal_7:any = this._map_2(this.context.noIncrementMinutes());
    this._NgClass_15_3.check_ngClass(currVal_7,throwOnChange,false);
    this._NgClass_15_3.detectChangesInternal(this,this._el_15,throwOnChange);
    const currVal_8:any = this.context.showMeridian;
    this._NgIf_18_6.check_ngIf(currVal_8,throwOnChange,false);
    this._NgIf_18_6.detectChangesInternal(this,this._anchor_18,throwOnChange);
    const currVal_9:any = 'form-group';
    this._NgClass_23_3.check_klass(currVal_9,throwOnChange,false);
    const currVal_10:any = this._map_3(this.context.invalidHours);
    this._NgClass_23_3.check_ngClass(currVal_10,throwOnChange,false);
    this._NgClass_23_3.detectChangesInternal(this,this._el_23,throwOnChange);
    this._DefaultValueAccessor_25_3.detectChangesInternal(this,this._el_25,throwOnChange);
    const currVal_17:any = '2';
    this._MaxLengthValidator_25_4.check_maxlength(currVal_17,throwOnChange,false);
    this._MaxLengthValidator_25_4.detectChangesInternal(this,this._el_25,throwOnChange);
    const currVal_19:any = this.context.hours;
    this._NgModel_25_7.check_model(currVal_19,throwOnChange,false);
    this._NgModel_25_7.detectChangesInternal(this,this._el_25,throwOnChange);
    this._NgControlStatus_25_9.detectChangesInternal(this,this._el_25,throwOnChange);
    const currVal_26:any = 'form-group';
    this._NgClass_31_3.check_klass(currVal_26,throwOnChange,false);
    const currVal_27:any = this._map_4(this.context.invalidMinutes);
    this._NgClass_31_3.check_ngClass(currVal_27,throwOnChange,false);
    this._NgClass_31_3.detectChangesInternal(this,this._el_31,throwOnChange);
    this._DefaultValueAccessor_33_3.detectChangesInternal(this,this._el_33,throwOnChange);
    const currVal_34:any = '2';
    this._MaxLengthValidator_33_4.check_maxlength(currVal_34,throwOnChange,false);
    this._MaxLengthValidator_33_4.detectChangesInternal(this,this._el_33,throwOnChange);
    const currVal_36:any = this.context.minutes;
    this._NgModel_33_7.check_model(currVal_36,throwOnChange,false);
    this._NgModel_33_7.detectChangesInternal(this,this._el_33,throwOnChange);
    this._NgControlStatus_33_9.detectChangesInternal(this,this._el_33,throwOnChange);
    const currVal_43:any = this.context.showMeridian;
    this._NgIf_36_6.check_ngIf(currVal_43,throwOnChange,false);
    this._NgIf_36_6.detectChangesInternal(this,this._anchor_36,throwOnChange);
    const currVal_44:any = 'text-center';
    this._NgClass_39_3.check_klass(currVal_44,throwOnChange,false);
    const currVal_45:any = this._map_5((!this.context.showSpinners || this.context.readonlyInput));
    this._NgClass_39_3.check_ngClass(currVal_45,throwOnChange,false);
    this._NgClass_39_3.detectChangesInternal(this,this._el_39,throwOnChange);
    const currVal_47:any = 'btn btn-link';
    this._NgClass_42_3.check_klass(currVal_47,throwOnChange,false);
    const currVal_48:any = this._map_6(this.context.noDecrementHours());
    this._NgClass_42_3.check_ngClass(currVal_48,throwOnChange,false);
    this._NgClass_42_3.detectChangesInternal(this,this._el_42,throwOnChange);
    const currVal_50:any = 'btn btn-link';
    this._NgClass_49_3.check_klass(currVal_50,throwOnChange,false);
    const currVal_51:any = this._map_7(this.context.noDecrementMinutes());
    this._NgClass_49_3.check_ngClass(currVal_51,throwOnChange,false);
    this._NgClass_49_3.detectChangesInternal(this,this._el_49,throwOnChange);
    const currVal_52:any = this.context.showMeridian;
    this._NgIf_52_6.check_ngIf(currVal_52,throwOnChange,false);
    this._NgIf_52_6.detectChangesInternal(this,this._anchor_52,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_16:any = this.context.readonlyInput;
    if (import2.checkBinding(throwOnChange,this._expr_16,currVal_16)) {
      this.renderer.setElementProperty(this._el_25,'readOnly',currVal_16);
      this._expr_16 = currVal_16;
    }
    const currVal_18:any = (this._MaxLengthValidator_25_4.context.maxlength? this._MaxLengthValidator_25_4.context.maxlength: (null as any));
    if (import2.checkBinding(throwOnChange,this._expr_18,currVal_18)) {
      this.renderer.setElementAttribute(this._el_25,'maxlength',((currVal_18 == (null as any))? (null as any): currVal_18.toString()));
      this._expr_18 = currVal_18;
    }
    const currVal_20:any = this._NgControlStatus_25_9.context.ngClassUntouched;
    if (import2.checkBinding(throwOnChange,this._expr_20,currVal_20)) {
      this.renderer.setElementClass(this._el_25,'ng-untouched',currVal_20);
      this._expr_20 = currVal_20;
    }
    const currVal_21:any = this._NgControlStatus_25_9.context.ngClassTouched;
    if (import2.checkBinding(throwOnChange,this._expr_21,currVal_21)) {
      this.renderer.setElementClass(this._el_25,'ng-touched',currVal_21);
      this._expr_21 = currVal_21;
    }
    const currVal_22:any = this._NgControlStatus_25_9.context.ngClassPristine;
    if (import2.checkBinding(throwOnChange,this._expr_22,currVal_22)) {
      this.renderer.setElementClass(this._el_25,'ng-pristine',currVal_22);
      this._expr_22 = currVal_22;
    }
    const currVal_23:any = this._NgControlStatus_25_9.context.ngClassDirty;
    if (import2.checkBinding(throwOnChange,this._expr_23,currVal_23)) {
      this.renderer.setElementClass(this._el_25,'ng-dirty',currVal_23);
      this._expr_23 = currVal_23;
    }
    const currVal_24:any = this._NgControlStatus_25_9.context.ngClassValid;
    if (import2.checkBinding(throwOnChange,this._expr_24,currVal_24)) {
      this.renderer.setElementClass(this._el_25,'ng-valid',currVal_24);
      this._expr_24 = currVal_24;
    }
    const currVal_25:any = this._NgControlStatus_25_9.context.ngClassInvalid;
    if (import2.checkBinding(throwOnChange,this._expr_25,currVal_25)) {
      this.renderer.setElementClass(this._el_25,'ng-invalid',currVal_25);
      this._expr_25 = currVal_25;
    }
    const currVal_33:any = this.context.readonlyInput;
    if (import2.checkBinding(throwOnChange,this._expr_33,currVal_33)) {
      this.renderer.setElementProperty(this._el_33,'readOnly',currVal_33);
      this._expr_33 = currVal_33;
    }
    const currVal_35:any = (this._MaxLengthValidator_33_4.context.maxlength? this._MaxLengthValidator_33_4.context.maxlength: (null as any));
    if (import2.checkBinding(throwOnChange,this._expr_35,currVal_35)) {
      this.renderer.setElementAttribute(this._el_33,'maxlength',((currVal_35 == (null as any))? (null as any): currVal_35.toString()));
      this._expr_35 = currVal_35;
    }
    const currVal_37:any = this._NgControlStatus_33_9.context.ngClassUntouched;
    if (import2.checkBinding(throwOnChange,this._expr_37,currVal_37)) {
      this.renderer.setElementClass(this._el_33,'ng-untouched',currVal_37);
      this._expr_37 = currVal_37;
    }
    const currVal_38:any = this._NgControlStatus_33_9.context.ngClassTouched;
    if (import2.checkBinding(throwOnChange,this._expr_38,currVal_38)) {
      this.renderer.setElementClass(this._el_33,'ng-touched',currVal_38);
      this._expr_38 = currVal_38;
    }
    const currVal_39:any = this._NgControlStatus_33_9.context.ngClassPristine;
    if (import2.checkBinding(throwOnChange,this._expr_39,currVal_39)) {
      this.renderer.setElementClass(this._el_33,'ng-pristine',currVal_39);
      this._expr_39 = currVal_39;
    }
    const currVal_40:any = this._NgControlStatus_33_9.context.ngClassDirty;
    if (import2.checkBinding(throwOnChange,this._expr_40,currVal_40)) {
      this.renderer.setElementClass(this._el_33,'ng-dirty',currVal_40);
      this._expr_40 = currVal_40;
    }
    const currVal_41:any = this._NgControlStatus_33_9.context.ngClassValid;
    if (import2.checkBinding(throwOnChange,this._expr_41,currVal_41)) {
      this.renderer.setElementClass(this._el_33,'ng-valid',currVal_41);
      this._expr_41 = currVal_41;
    }
    const currVal_42:any = this._NgControlStatus_33_9.context.ngClassInvalid;
    if (import2.checkBinding(throwOnChange,this._expr_42,currVal_42)) {
      this.renderer.setElementClass(this._el_33,'ng-invalid',currVal_42);
      this._expr_42 = currVal_42;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._NgModel_25_7.context.ngOnDestroy();
    this._NgModel_33_7.context.ngOnDestroy();
  }
  private _handle_click_8_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.incrementHours()) !== false);
    return (true && pd_0);
  }
  private _handle_click_15_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.incrementMinutes()) !== false);
    return (true && pd_0);
  }
  private _handle_ngModelChange_25_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>(this.context.hours = $event)) !== false);
    return (true && pd_0);
  }
  private _handle_change_25_1($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.updateHours()) !== false);
    return (true && pd_0);
  }
  private _handle_blur_25_2($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.hoursOnBlur($event)) !== false);
    const pd_1:any = ((<any>this._DefaultValueAccessor_25_3.context.onTouched()) !== false);
    return ((true && pd_0) && pd_1);
  }
  private _handle_input_25_3($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this._DefaultValueAccessor_25_3.context.onChange($event.target.value)) !== false);
    return (true && pd_0);
  }
  private _handle_ngModelChange_33_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>(this.context.minutes = $event)) !== false);
    return (true && pd_0);
  }
  private _handle_change_33_1($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.updateMinutes()) !== false);
    return (true && pd_0);
  }
  private _handle_blur_33_2($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.minutesOnBlur($event)) !== false);
    const pd_1:any = ((<any>this._DefaultValueAccessor_33_3.context.onTouched()) !== false);
    return ((true && pd_0) && pd_1);
  }
  private _handle_input_33_3($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this._DefaultValueAccessor_33_3.context.onChange($event.target.value)) !== false);
    return (true && pd_0);
  }
  private _handle_click_42_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.decrementHours()) !== false);
    return (true && pd_0);
  }
  private _handle_click_49_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.decrementMinutes()) !== false);
    return (true && pd_0);
  }
}
export function viewFactory_TimepickerComponent0(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<import0.TimepickerComponent> {
  if ((renderType_TimepickerComponent === (null as any))) { (renderType_TimepickerComponent = viewUtils.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_TimepickerComponent,{})); }
  return new _View_TimepickerComponent0(viewUtils,parentInjector,declarationEl);
}
class _View_TimepickerComponent1 extends import3.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import12.Wrapper_NgClass;
  _map_0:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_TimepickerComponent1,renderType_TimepickerComponent,import8.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'td',(null as any));
    this._NgClass_0_3 = new import12.Wrapper_NgClass(this.parent.parentInjector.get(import18.IterableDiffers),this.parent.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_0),this.renderer);
    this._map_0 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0:any = this._map_0(!this.parent.context.showMeridian);
    this._NgClass_0_3.check_ngClass(currVal_0,throwOnChange,false);
    this._NgClass_0_3.detectChangesInternal(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_TimepickerComponent1(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  return new _View_TimepickerComponent1(viewUtils,parentInjector,declarationEl);
}
class _View_TimepickerComponent2 extends import3.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import12.Wrapper_NgClass;
  _el_1:any;
  _NgClass_1_3:import12.Wrapper_NgClass;
  _text_2:any;
  _map_0:any;
  _map_1:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_TimepickerComponent2,renderType_TimepickerComponent,import8.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'td',(null as any));
    this._NgClass_0_3 = new import12.Wrapper_NgClass(this.parent.parentInjector.get(import18.IterableDiffers),this.parent.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_0),this.renderer);
    this._el_1 = this.renderer.createElement(this._el_0,'button',(null as any));
    this.renderer.setElementAttribute(this._el_1,'class','btn btn-default text-center');
    this.renderer.setElementAttribute(this._el_1,'type','button');
    this._NgClass_1_3 = new import12.Wrapper_NgClass(this.parent.parentInjector.get(import18.IterableDiffers),this.parent.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'',(null as any));
    this._map_0 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
    var disposable_0:Function = this.renderer.listen(this._el_1,'click',this.eventHandler(this._handle_click_1_0.bind(this)));
    this._map_1 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this._expr_4 = import1.UNINITIALIZED;
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._el_1,
      this._text_2
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._NgClass_1_3.context; }
    if (((token === import22.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0:any = this._map_0(!this.parent.context.showMeridian);
    this._NgClass_0_3.check_ngClass(currVal_0,throwOnChange,false);
    this._NgClass_0_3.detectChangesInternal(this,this._el_0,throwOnChange);
    const currVal_2:any = 'btn btn-default text-center';
    this._NgClass_1_3.check_klass(currVal_2,throwOnChange,false);
    const currVal_3:any = this._map_1((this.parent.context.noToggleMeridian() || this.parent.context.readonlyInput));
    this._NgClass_1_3.check_ngClass(currVal_3,throwOnChange,false);
    this._NgClass_1_3.detectChangesInternal(this,this._el_1,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_4:any = import2.interpolate(1,'',this.parent.context.meridian,'');
    if (import2.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setText(this._text_2,currVal_4);
      this._expr_4 = currVal_4;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_1_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.parent.context.toggleMeridian()) !== false);
    return (true && pd_0);
  }
}
function viewFactory_TimepickerComponent2(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  return new _View_TimepickerComponent2(viewUtils,parentInjector,declarationEl);
}
class _View_TimepickerComponent3 extends import3.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import12.Wrapper_NgClass;
  _map_0:any;
  constructor(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement) {
    super(_View_TimepickerComponent3,renderType_TimepickerComponent,import8.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import5.AppElement {
    this._el_0 = this.renderer.createElement((null as any),'td',(null as any));
    this._NgClass_0_3 = new import12.Wrapper_NgClass(this.parent.parentInjector.get(import18.IterableDiffers),this.parent.parentInjector.get(import19.KeyValueDiffers),new import20.ElementRef(this._el_0),this.renderer);
    this._map_0 = import2.pureProxy1((p0:any):{[key: string]:any} => {
      return {hidden: p0};
    });
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import22.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0:any = this._map_0(!this.parent.context.showMeridian);
    this._NgClass_0_3.check_ngClass(currVal_0,throwOnChange,false);
    this._NgClass_0_3.detectChangesInternal(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_TimepickerComponent3(viewUtils:import2.ViewUtils,parentInjector:import7.Injector,declarationEl:import5.AppElement):import3.AppView<any> {
  return new _View_TimepickerComponent3(viewUtils,parentInjector,declarationEl);
}
