// @angular/common/src/directives/ng_template_outlet.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
import * as import0 from '@angular/common/src/directives/ng_template_outlet';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
export var Wrapper_NgTemplateOutlet = (function () {
    function Wrapper_NgTemplateOutlet(p0) {
        this.changed = false;
        this.changes = {};
        this._ngOutletContext = import1.UNINITIALIZED;
        this._ngTemplateOutlet = import1.UNINITIALIZED;
        this.context = new import0.NgTemplateOutlet(p0);
    }
    Wrapper_NgTemplateOutlet.prototype.check_ngOutletContext = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._ngOutletContext, currValue))) {
            this.changed = true;
            this.context.ngOutletContext = currValue;
            this.changes['ngOutletContext'] = new import1.SimpleChange(this._ngOutletContext, currValue);
            this._ngOutletContext = currValue;
        }
    };
    Wrapper_NgTemplateOutlet.prototype.check_ngTemplateOutlet = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._ngTemplateOutlet, currValue))) {
            this.changed = true;
            this.context.ngTemplateOutlet = currValue;
            this.changes['ngTemplateOutlet'] = new import1.SimpleChange(this._ngTemplateOutlet, currValue);
            this._ngTemplateOutlet = currValue;
        }
    };
    Wrapper_NgTemplateOutlet.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        if (!throwOnChange) {
            if (changed) {
                this.context.ngOnChanges(this.changes);
                this.changes = {};
            }
        }
        return changed;
    };
    return Wrapper_NgTemplateOutlet;
}());
