// ng2-bootstrap/components/modal/modal-backdrop.component.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from 'ng2-bootstrap/components/modal/modal-backdrop.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/element';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/element_ref';
import * as import9 from '@angular/core/src/metadata/view';
import * as import10 from '@angular/core/src/linker/component_factory';
export var Wrapper_ModalBackdropComponent = (function () {
    function Wrapper_ModalBackdropComponent(p0, p1, p2) {
        this.changed = false;
        this.context = new import0.ModalBackdropComponent(p0, p1, p2);
    }
    Wrapper_ModalBackdropComponent.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_ModalBackdropComponent;
}());
var renderType_ModalBackdropComponent_Host = null;
var _View_ModalBackdropComponent_Host0 = (function (_super) {
    __extends(_View_ModalBackdropComponent_Host0, _super);
    function _View_ModalBackdropComponent_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_ModalBackdropComponent_Host0, renderType_ModalBackdropComponent_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_ModalBackdropComponent_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('bs-modal-backdrop', rootSelector, null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'modal-backdrop');
        this._appEl_0 = new import3.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_ModalBackdropComponent0(this.viewUtils, this.injector(0), this._appEl_0);
        this._ModalBackdropComponent_0_4 = new Wrapper_ModalBackdropComponent(this.parentInjector.get(import0.ModalBackdropOptions), new import8.ElementRef(this._el_0), this.renderer);
        this._appEl_0.initComponent(this._ModalBackdropComponent_0_4.context, [], compView_0);
        compView_0.create(this._ModalBackdropComponent_0_4.context, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_ModalBackdropComponent_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.ModalBackdropComponent) && (0 === requestNodeIndex))) {
            return this._ModalBackdropComponent_0_4.context;
        }
        return notFoundResult;
    };
    _View_ModalBackdropComponent_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this._ModalBackdropComponent_0_4.detectChangesInternal(this, this._el_0, throwOnChange);
        this.detectContentChildrenChanges(throwOnChange);
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_ModalBackdropComponent_Host0;
}(import1.AppView));
function viewFactory_ModalBackdropComponent_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ModalBackdropComponent_Host === null)) {
        (renderType_ModalBackdropComponent_Host = viewUtils.createRenderComponentType('', 0, import9.ViewEncapsulation.None, [], {}));
    }
    return new _View_ModalBackdropComponent_Host0(viewUtils, parentInjector, declarationEl);
}
export var ModalBackdropComponentNgFactory = new import10.ComponentFactory('bs-modal-backdrop', viewFactory_ModalBackdropComponent_Host0, import0.ModalBackdropComponent);
var styles_ModalBackdropComponent = [];
var renderType_ModalBackdropComponent = null;
var _View_ModalBackdropComponent0 = (function (_super) {
    __extends(_View_ModalBackdropComponent0, _super);
    function _View_ModalBackdropComponent0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_ModalBackdropComponent0, renderType_ModalBackdropComponent, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_ModalBackdropComponent0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this.init([], [], [], []);
        return null;
    };
    return _View_ModalBackdropComponent0;
}(import1.AppView));
export function viewFactory_ModalBackdropComponent0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ModalBackdropComponent === null)) {
        (renderType_ModalBackdropComponent = viewUtils.createRenderComponentType('', 0, import9.ViewEncapsulation.None, styles_ModalBackdropComponent, {}));
    }
    return new _View_ModalBackdropComponent0(viewUtils, parentInjector, declarationEl);
}
